### DISCUSSION TITLE
Abp Community Input: Best Practices for Third-Party App Integration and App Store Development

### DISCUSSION THREAD

**Discussion by ahmadalmunajjed** [abpframework/abp#16948] [Ideas]
Hey ABP community! 👋

I'm currently working on a SaaS project using our beloved Abp framework, and I've hit an exciting stage - I'm considering the implementation of an "app store" or "plugin store" to enhance our platform's extensibility.

I believe many of you may have already walked this path, and your insights could be incredibly valuable as I navigate through this process. Here are the areas where I'm seeking your advice:

1️⃣ **Developer Management:** What strategies have you used to allow developers to manage and submit their apps to your platform?
2️⃣ **Installation & Authentication:** What were your best practices for the installation and authentication process of third-party apps?
3️⃣ **API Communication:**
- How did you establish communication from third-party apps to your platform API using RESTful APIs?
- How did you manage communication from your API to third-party apps, possibly through Webhooks?

4️⃣ **Building the Marketplace:** Any tips or experiences on how to effectively build and manage the marketplace/storefront for these apps?

I'm looking forward to hearing your thoughts, learning from your experiences, and maybe even hearing about some challenges you've faced. Thanks a ton in advance for your insights!

**Comment by rashedjawda** [comment]
I like the ideas found in Umbraco Marketplace https://marketplace.umbraco.com/ 
it also open inside you umbraco installation

**Reply by tonyw54** [reply]
Umbraco looks good. More than just the marketplace. I just wish there were a way to run ABP, but use Umbraco as the CMS, instead of cms-kit. Umbraco is way, way better.
